packages/flutter_tools/bin/podhelper.rb

```diff
@@ -19,6 +19,29 @@ def flutter_ios_podfile_setup; end
 # Same as flutter_ios_podfile_setup for macOS.
 def flutter_macos_podfile_setup; end
 
+# Determine whether the target depends on Flutter (including transitive dependency)
+def is_depend_on_flutter(target, platform)
+  if platform == 'ios'
+    engine_pod_name = 'Flutter'
+  elsif platform == 'macos'
+    engine_pod_name = 'FlutterMacOS'
```

Comment:
What is the goal passing `ios` or `macos` only to map back to the framework name, rather than passing the framework name directly? It seems like this just adds complexity.